ntbs-service/DataMigration/ImportValidator.cs

```diff
@@ -101,21 +101,43 @@ public ImportValidator(IImportLogger logger, IReferenceDataRepository referenceD
 
         private static void TrimAndCleanLongFormNotes(Notification notification)
         {
-            CleanContainsNoTabsProperties(notification.ClinicalDetails);
-            notification.SocialContextAddresses.ForEach(CleanContainsNoTabsProperties);
-            notification.SocialContextVenues.ForEach(CleanContainsNoTabsProperties);
-            notification.TreatmentEvents.ForEach(CleanContainsNoTabsProperties);
-            notification.MBovisDetails.MBovisExposureToKnownCases.ForEach(CleanContainsNoTabsProperties);
-            notification.MBovisDetails.MBovisOccupationExposures.ForEach(CleanContainsNoTabsProperties);
-            notification.MBovisDetails.MBovisAnimalExposures.ForEach(CleanContainsNoTabsProperties);
-            notification.MBovisDetails.MBovisUnpasteurisedMilkConsumptions.ForEach(CleanContainsNoTabsProperties);
+            TrimAndCleanStringProperties(notification.PatientDetails);
+            TrimAndCleanStringProperties(notification.ClinicalDetails);
+            TrimAndCleanStringProperties(notification.HospitalDetails);
+            TrimAndCleanStringProperties(notification.PreviousTbHistory); // ?
+            TrimAndCleanStringProperties(notification.ComorbidityDetails); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors?.RiskFactorHomelessness); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors?.RiskFactorDrugs); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors?.RiskFactorImprisonment); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors?.RiskFactorSmoking); // ?
+            TrimAndCleanStringProperties(notification.ImmunosuppressionDetails);
+            TrimAndCleanStringProperties(notification.TravelDetails); // ?
+            TrimAndCleanStringProperties(notification.VisitorDetails); // ?
+            TrimAndCleanStringProperties(notification.DenotificationDetails); // untested
+            TrimAndCleanStringProperties(notification.ComorbidityDetails); // ?
+            TrimAndCleanStringProperties(notification.MDRDetails);
+            TrimAndCleanStringProperties(notification.Group); // ? untested
+            TrimAndCleanStringProperties(notification.TestData); // ?
+            TrimAndCleanStringProperties(notification.MBovisDetails); // ?
+            TrimAndCleanStringProperties(notification.DrugResistanceProfile);
```

Comment:
There are a lot of child properties getting enumerated which isn't ideal but still probably the best approach I think. 

A question I have here is: several of the child properties (marked `// ?`) are properties that don't have any `string` properties themselves, so could be safely omitted at the moment. However, in future if one of these classes were to gain some `string` properties they wouldn't be cleaned until this line was added back in, so there is an argument for including them (though not a fully consistent one because if a new class/table was added then you would need to edit here). Should I be included the `// ?` lines or not.

(Aside: `// untested` are items that currently are `null` in the `ImportValidatorTest` so need to be instantiated. It is just a temporary comment and not of use to the PR)

Reply:
That's a good idea! I've added two types of attributes, `ValidationChild` for object children like `ClinicalDetails` in `Notification`, and `ValidationChildEnumerable`, for enumerable children like `TreatmentEvents` in `Notification`